aspnet/client-side/bundling-and-minification.rst

```diff
@@ -1,10 +1,177 @@
+
+.. image:: bundling-and-minification/_static/03-MinTiming.png
+
+Minification
+------------
+Minification performs a variety of different code optimizations to reduce the size of requested assets (such as CSS, image, JavaScript files). Common results of minification include removing unnecessary white space and comments, and shortening variable names to one character. 
+
+Consider the following JavaScript function:
+
+.. code-block:: javascript
+
+	AddAltToImg = function (imageTagAndImageID, imageContext) {
+		///<signature>
+		///<summary> Adds an alt tab to the image
+		// </summary>
+		//<param name="imgElement" type="String">The image selector.</param>
+		//<param name="ContextForImage" type="String">The image context.</param>
+		///</signature>
+		var imageElement = $(imageTagAndImageID, imageContext);
+		imageElement.attr('alt', imageElement.attr('id').replace(/ID/, ''));
+	}
+
+After minification, the function is reduced to the following:
+
+.. code-block:: javascript
+
+	AddAltToImg=function(t,a){var r=$(t,a);r.attr("alt",r.attr("id").replace(/ID/,""))};
+	
+In addition to removing the comments and unnecessary whitespace, the following parameters and variable names were renamed (shortened) as follows:
+
+==================  =======  
+Original            Renamed      
+==================  =======  
+imageTagAndImageID  t  
+imageContext        a  
+imageElement        r   
+==================  =======  
+
+Impact of Bundling and Minification
+-----------------------------------
+
+The following table shows several important differences between listing all the assets individually and using bundling and minification based on the above *About* page.
+
+==================  ==========  ============  ============  
+Action              With B/M    Without B/M   Change    
+==================  ==========  ============  ============  
+File Requests       7           18            157%
+KB Transferred      156         264.68        70%
+Load Time (MS)      885         2360          167%  
+==================  ==========  ============  ============  
+
+The bytes sent had a significant reduction with bundling as browsers are fairly verbose with the HTTP headers that they apply on requests. The load time shows a big improvement, however this example was run locally. You will get greater gains in performance when using bundling and minification with assets transferred over a network. To simulate the transfer time of a slow network, you can use the `Fiddler <http://www.telerik.com/fiddler>`_ tool. Within Fiddler, select the **Fiddler Rules** menu, **Performance** > **Simulate Modem Speeds**.
+
+Controlling Bundling and Minification
+-------------------------------------
+
+When working with an app created using the **ASP.NET 5 Web Application** template in Visual Studio, bundling and minification is enabled by setting the ``Hosting:Environment`` value to ``Production``. 
```

Comment:
This jumps a bit too fast into the VS specific mechanics. I think there should be an intro sentence here that says you generally want to use the bundled and minified files only in production. During development you want to original files so your app is easier to debug. Then explain that you can use ASP.NET Core to detect which environment you are in and link the corresponding files. Finish up with your example below of using the environments tag helper to do this. You can then talk about how VS makes it easy to build the bundled and minified files and (using the Task Runner Explorer) and provides templates with the related code.